R/teal_data_module.R

```diff
@@ -61,15 +61,71 @@
 #' @export
 teal_data_module <- function(ui, server, label = "data module", once = TRUE) {
   checkmate::assert_function(ui, args = "id", nargs = 1)
-  checkmate::assert(
-    checkmate::check_function(server, args = "id", nargs = 1),
-    # todo: allow for teal_data_module$server to have 'data' argument or break this in teal_transformer_module
-    checkmate::check_function(server, args = c("id", "data"), nargs = 2)
-  )
+  checkmate::check_function(server, args = "id", nargs = 1)
   structure(
     list(ui = ui, server = server),
     label = label,
     class = "teal_data_module",
     once = once
   )
 }
+
+#' Data module for `teal` transformers.
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' Create a `teal_data_module` object for custom transformation of data for pre-processing before passing the data into the module.
+#'
+#' @details
+#' `teal_transform_module` creates a `teal_data_module` object to transform the data in a `teal` application
+#' after it has been passed through the filter panel and before it is passed to the modules which uses the transform module.
```

Comment:
"modules" is too ambiguous, please use "server instance of a [teal_module()]".
```suggestion
#' `teal_transform_module` creates a `teal_data_module` object to transform a filtered data
#' before handing over to the server of a [teal_module()].
```

Reply:
```suggestion
#' `teal_transform_module` creates a `teal_data_module` object to transform data in a `teal` application.
#' This transformation happens after the data has passed through the filtering activity in teal. The transformed data is then sent to the server of the [teal_module()].
```
Another alternative